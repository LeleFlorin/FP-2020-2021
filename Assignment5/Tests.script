:l Solutions.hs

take 5 passwords
take 3 (drop 10 passwords)
take 5 (drop 10000 passwords)

validatePassword "123"
validatePassword "abc"
validatePassword "MyStr0ngPassword2"
validatePassword "abcDE12"
validatePassword "Abcd1234"
validatePassword "abcd1234"
validatePassword "abcd12A"

validateEmail "johnny@example.com"
validateEmail "johnny@.com"
validateEmail "johnny.com"
validateEmail "johnny@domain@.com"
validateEmail "x@domain.com"
validateEmail "johnny.john@domain.com"
validateEmail "johnny.john@domain.example.com"
validateEmail "last.da.first@didatec.utcluj.mail.com"
validateEmail "last.da.first@didatec.utcluj.mail.ro"
validateEmail "last.da.first@didatec..utcluj.mail.com"
validateEmail "last.da.first@didatec.utcluj.mail..com"
validateEmail "last.da.first@@didatec.utcluj.mail.ro"
