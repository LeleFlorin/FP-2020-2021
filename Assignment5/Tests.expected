["00000000","00000001","00000002","00000003","00000004"]
["0000000a","0000000b","0000000c"]
["000002Bi","000002Bj","000002Bk","000002Bl","000002Bm"]

Nothing
Nothing
Just (Password "MyStr0ngPassword2")
Nothing
Just (Password "Abcd1234")
Nothing
Nothing

Just (Email {username = "johnny", domain = "example.com"})
Nothing
Nothing
Nothing
Nothing
Just (Email {username = "johnny.john", domain = "domain.com"})
Just (Email {username = "johnny.john", domain = "domain.example.com"})
Just (Email {username = "last.da.first", domain = "didatec.utcluj.mail.com"})
Just (Email {username = "last.da.first", domain = "didatec.utcluj.mail.ro"})
Nothing
Nothing
Nothing
